common/src/Model/BrushFaceSnapshot.cpp

```diff
@@ -19,22 +19,30 @@
 
 #include "BrushFaceSnapshot.h"
 
+#include "Model/Brush.h"
+
 namespace TrenchBroom {
     namespace Model {
         BrushFaceSnapshot::BrushFaceSnapshot(BrushFace* face, TexCoordSystem* coordSystem) :
-        m_face(face),
+        m_brush(face->brush()),
+        m_faceBoundary(face->boundary()),
         m_attribs(face->attribs().takeSnapshot()),
-        m_coordSystem(coordSystem->takeSnapshot()) {}
+        m_coordSystemSnapshot(coordSystem->takeSnapshot()) {
+            ensure(m_brush != nullptr, "need a brush to snapshot a face");
+        }
         
         BrushFaceSnapshot::~BrushFaceSnapshot() {
-            delete m_coordSystem;
-            m_coordSystem = NULL;
+            delete m_coordSystemSnapshot;
+            m_coordSystemSnapshot = NULL;
         }
 
         void BrushFaceSnapshot::restore() {
-            m_face->setAttribs(m_attribs);
-            if (m_coordSystem != NULL)
-                m_coordSystem->restore();
+            BrushFace* face = m_brush->findFace(m_faceBoundary);
```

Comment:
Shouldn't it be enough to find the face by its normal? Not sure if that's better or not, just thinking out load.